src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,64 @@
+package core.basesyntax;
+
+import core.basesyntax.model.Figure;
+import java.util.Random;
+
+public class FigureSupplier {
+
+    private static final String DEFAULT_COLOR = "WHITE";
+    private static final int RANDOM_LIMIT_NUMBER = 100;
+    private static final int DEFAULT_RADIUS_NUMBER = 10;
+
+    private final Random rand = new Random();
+    private final ColorSupplier colorSupplier = new ColorSupplier();
+
+    private enum FigureType {
+        SQUARE,
+        RECTANGLE,
+        RIGHT_TRIANGLE,
+        CIRCLE,
+        ISOSCELES_TRAPEZOID
+    }
+
+    public Figure getRandomFigure() {
+        String type = getRandomType();
+        String color = colorSupplier.getRandomColor();
+
+        if (type.equals(FigureType.SQUARE.name())) {
+            double side = rand.nextInt(RANDOM_LIMIT_NUMBER);
+
+            return new Square(color, side);
```

Comment:
lets add private methods to get each random figure, this way your code would be easier to read and understand